lib/services/elasticsearch.js

```diff
@@ -301,60 +301,55 @@ class ElasticSearch extends Service {
 
   /**
    * Returns the list of documents matching the ids given in the body param
-   * NB: Due to internal Kuzzle mechanism, can only be called on a single index/collection,
-   * using the body { ids: [.. } syntax.
+   * NB: Due to internal Kuzzle mechanism, can only be called on a single
+   * index/collection, using the body { ids: [.. } syntax.
    *
    * @param {String} index - Index name
    * @param {String} collection - Collection name
    * @param {Array.<String>} ids - Document IDs
    *
    * @returns {Promise.<Object>} { items: [ _id, _source, _version ], errors }
    */
-  mGet (index, collection, ids) {
+  async mGet (index, collection, ids) {
     if (ids.length === 0) {
-      return Bluebird.resolve({ item: [], errors: [] });
+      return { item: [], errors: [] };
     }
 
     const
       esIndex = this._getESIndex(index, collection),
       esRequest = {
-        body: { docs: [] }
+        body: {
+          docs: ids.map(_id => ({ _index: esIndex, _id }))
+        }
       };
 
-    for (const _id of ids) {
-      esRequest.body.docs.push({
-        _index: esIndex,
-        _id
-      });
+    debug('Multi-get documents: %o', esRequest);
+
+    let body;
+
+    try {
+      ({ body } = await this._client.mget(esRequest)); // NOSONAR
+    }
+    catch (e) {
+      throw this._esWrapper.formatESError(e);
     }
 
-    debug('Multi-get documents: %o', esRequest);
+    const
+      errors = [],
+      items = [];
 
-    return this._client.mget(esRequest)
-      .then(({ body }) => {
-        const
-          errors = [],
-          items = [];
-
-        for (const doc of body.docs) {
-          if (doc.found) {
-            items.push({
-              _id: doc._id,
-              _source: doc._source,
-              _version: doc._version
-            });
-          }
-          else {
-            errors.push(doc._id);
-          }
-        }
+    for (let i = 0; i < body.docs.length; i++) {
+      const { found, _id, _source, _version } = body.docs[i];
```

Comment:
I prefer to avoid destructuring in critical code areas because it's slower than simple access
```
access property x 101,799,727 ops/sec ±0.22% (95 runs sampled)
destructuring x 64,808,312 ops/sec ±0.32% (96 runs sampled)
```
https://gist.github.com/Aschen/f1c790fbfa0a8825bccd4fdb75f3becf

Also, in the particular case of creating an array from another array, `Array.map` is even faster than a `for (let i ...)` loop
```
   foreach x 60,450 ops/sec ±8.91% (87 runs sampled)
   for (const of) x 112,013 ops/sec ±1.97% (84 runs sampled)
   for (let i) x 113,350 ops/sec ±2.63% (82 runs sampled)
   map x 153,657 ops/sec ±4.45% (78 runs sampled)
```
https://gist.github.com/Aschen/6225c7bc9c67313f9eeb3571964984e5

Reply:
can't use array.map for that particular example though